appsec/src/extension/tags.c

```diff
@@ -315,6 +319,12 @@ void dd_tags_add_appsec_json_frag(zend_string *nonnull zstr)
     zend_llist_add_element(&_appsec_json_frags, &zstr);
 }
 
+void dd_tags_set_user_event_triggered(bool trigger_by_user)
```

Comment:
Do we need two variables to track a single action? i.e. we just want to track whether an event requiring all headers was produced.

Reply:
I have just refactored it